cmd/server/main_test.go

```diff
@@ -0,0 +1,50 @@
+package main
+
+import (
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"io"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+)
+
+func testRequest(t *testing.T, ts *httptest.Server, method, path string) (*http.Response, string) {
+	req, err := http.NewRequest(method, ts.URL+path, nil)
+	require.NoError(t, err)
+
+	resp, err := ts.Client().Do(req)
+	require.NoError(t, err)
+	defer resp.Body.Close()
+
+	respBody, err := io.ReadAll(resp.Body)
+	require.NoError(t, err)
+
+	return resp, string(respBody)
+}
+
+func TestRouter(t *testing.T) {
+	ts := httptest.NewServer(NewRouter())
+	defer ts.Close()
+
+	var testTable = []struct {
+		url    string
+		want   string
+		status int
+		method string
+	}{
+		{"/update/gauge/testGaugeMetricName/328479.927", "", http.StatusOK, "POST"},
+		{"/value/gauge/testGaugeMetricName", "328479.927", http.StatusOK, "GET"},
+		{"/update/counter/testMetricName/123", "", http.StatusOK, "POST"},
+		{"/update/wrongType/testMetricName/123", "", http.StatusBadRequest, "POST"},
+		{"/value/counter/testMetricName", "123", http.StatusOK, "GET"},
+		{"/value/counter/wrongTestMetricName", "", http.StatusNotFound, "GET"},
+		{"/", "testGaugeMetricName: 328479.927 \ntestMetricName: 123 \n", http.StatusOK, "GET"},
+	}
+	for _, v := range testTable {
+		resp, get := testRequest(t, ts, v.method, v.url)
+		assert.Equal(t, v.status, resp.StatusCode)
+		assert.Equal(t, v.want, get)
+		resp.Body.Close()
+	}
```

Comment:
Стоит добавить названия для табличных тестов, а так же использовать сабтесты `t.Run(...)` для выполнения каждого теста. Так же здорово было бы добавить немного больше информации о том, что ожидается и о том, что пошло не так. 

Вообще разрабочики Go не рекомендуют использовать сторонние assert библиотеки  так как это либо останавливает тест раньше времени (если вызовы утверждения t.Fatalf или panic), либо упускает интересную информацию о том, что тест сделал правильно. Это также заставляет assert библиотеку создавать совершенно новый подъязык вместо использования существующего языка программирования (самого Go). https://go.dev/wiki/TestComments

```suggestion
for _, v := range testTable {
		t.Run(v.name, func(t *testing.T) {
			resp, body := testRequest(t, ts, v.method, v.url)
			assert.Equal(t, v.status, resp.StatusCode, "Status code mismatch")
			assert.Equal(t, v.want, body, "Response body mismatch")
		})
	}
```